src/uct/ib/dc/dc_mlx5.h

```diff
@@ -197,11 +197,26 @@ typedef struct uct_dc_mlx5_ep_fc_entry {
 KHASH_MAP_INIT_INT64(uct_dc_mlx5_fc_hash, uct_dc_mlx5_ep_fc_entry_t);
 
 
+/* DCI pool
+ * there used same DCI stack to allocate and release DCI's:
+ * 
+ * +--------------+ +-------------+
+ * | to release   | | to allocate |
+ * +--------------+ +-------------+
```

Comment:
where are the allocated (used) DCIs in this picture?

Reply:
the picture is for 0..ndci-1
does it mean that num_dci_to_release+num_dci_to_allocate==ndci?

Reply:
no, there is special ```whole``` between dci_to_release and stack_top meaning that sum of these values are not same as ndci.

picture demonstrates structure of stack, dci_to_release pointer may have negative value meaning that no DCI's to release

Reply:
should be like this?
```
| to_release |  used | to_allocate |
```
or like this?
```
| used | to_release |  to_allocate |
```

Reply:
which kind of dci can be between "releast_top" and "stack_top" ? according to the picture their difference is 1, is it always the case?

Reply:
updated picture